offer-manager-app/src/main/groovy/life/qbic/portal/offermanager/dataresources/products/ProductsDbConnector.groovy

```diff
@@ -20,223 +21,279 @@ import java.sql.SQLException
-        productItems << item
-      }
+
+    /**
+     * Returns the product identifying running number given a productId
+     *
+     * @param productId String of productId stored in the DB e.g. "DS_1"
+     * @return identifier String of the iterative identifying part of the productId
+     */
+    static String parseProductId(String productId) {
+        def splitId = productId.split("_")
+        // The first entry [0] contains the product type which is assigned automatically, no need to parse it.
+        String identifier = splitId[1]
+        return identifier
     }
-    return productItems
-  }
 
-  /**
-   * Class that encapsulates the available SQL queries.
-   */
-  private static class Queries {
 
     /**
-     * Query for all available products.
+     * Returns the product type of a product based on its implemented class
+     *
+     * @param product A product for which the type needs to be determined
+     * @return the type of the product or null
      */
-    final static String SELECT_ALL_PRODUCTS = "SELECT * FROM product"
+    static String getProductType(Product product) {
+        if (product instanceof Sequencing) return 'Sequencing'
+        if (product instanceof ProjectManagement) return 'Project Management'
+        if (product instanceof PrimaryAnalysis) return 'Primary Bioinformatics'
+        if (product instanceof SecondaryAnalysis) return 'Secondary Bioinformatics'
+        if (product instanceof DataStorage) return 'Data Storage'
+
+        return null
+    }
 
     /**
-     * Query for all items of an offer.
+     * Queries all items of an offer.
+     * @param offerPrimaryId The offer's primary key.
+     * @return A list of offer-associated product items.
      */
-    final static String SELECT_ALL_ITEMS_FOR_OFFER =
-            "SELECT * FROM productitem " +
-                    "LEFT JOIN product ON productitem.productId = product.id " +
-                    "WHERE offerId=?;"
+    List<ProductItem> getItemsForOffer(int offerPrimaryId) {
+        List<ProductItem> productItems = []
+        Connection connection = provider.connect()
+        connection.withCloseable {
+            PreparedStatement statement = it.prepareStatement(Queries.SELECT_ALL_ITEMS_FOR_OFFER)
+            statement.setInt(1, offerPrimaryId)
+            ResultSet result = statement.executeQuery()
+            while (result.next()) {
+                Product product = rowResultToProduct(SqlExtensions.toRowResult(result))
+                double quantity = result.getDouble("quantity")
+                ProductItem item = new ProductItem(quantity, product)
+                productItems << item
+            }
+        }
+        return productItems
+    }
+
+    @Override
+    Optional<Product> fetch(ProductId productId) throws DatabaseQueryException {
+        Connection connection = provider.connect()
+        String query = Queries.SELECT_ALL_PRODUCTS + " WHERE productId=?"
```

Comment:
```suggestion
        String query = Queries.SELECT_ALL_PRODUCTS + " WHERE productId=?"
```
I suspect that this won't work. In the DB the productId is an `int(11)` we would need to change that

Reply:
I tested it out and it worked

Reply:
Ah I was looking at a `productitem#product_id` instead of the `product#productId`